### PR TITLE
Bump to lts-17

### LINKED ISSUE #35: Stop using aeson-options
# Clarification and motivation

`aeson-options` package is going to be deprecated: https://github.com/serokell/aeson-options/pull/9

But we depend on it.

# Acceptance criteria

`aeson-options` is not in the list of dependencies anymore.

### PR DESCRIPTION
**Pull request by Martoon-00**
## Description

Use `lts-17.3`.

## Related issue(s)

Resolves #35.

## :white_check_mark: Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](/README.md)
    - Haddock

- Public contracts
  - [x] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [x] I added an entry to the [changelog](/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](docs/code-style.md).

### CODE REVIEW

**Comment by Martoon-00** [comment]
At this point, I get some error about missing `ghc8104` in the CI build. @balsoft @zhenyavinogradov would appreciate your help.

**Comment by zhenyavinogradov** [comment]
I've updated haskell.nix and nixpkgs to the newer versions that we use in other repos, now it builds

File: src/Xrefcheck/Config.hs [hunk 1]
```diff
@@ -7,18 +7,18 @@
...
-import Data.Aeson.Options (defaultOptions)
 import Data.Aeson.TH (deriveFromJSON)
 import Data.Yaml (FromJSON (..), decodeEither', prettyPrintParseException, withText)
 import Instances.TH.Lift ()
 
 -- FIXME: Use </> from System.FilePath
 -- </> from Posix is used only because we cross-compile to Windows and \ doesn't work on Linux
-import System.FilePath.Posix ((</>))
 import Data.FileEmbed (embedFile)
+import System.FilePath.Posix ((</>))
```
Comment (zhenyavinogradov) [inline src/Xrefcheck/Config.hs:17]: The comment above seems misplaced now

**Review by zhenyavinogradov — APPROVED** [review]
Haskell part should probably be reviewed by someone else, but everything else looks good

**Comment by Martoon-00** [comment]
Great, thanks :+1:

File: src/Xrefcheck/Config.hs (hunk 1 above)
Comment (Martoon-00) [reply src/Xrefcheck/Config.hs:17]: Oh, good catch. A bit sad that such things occur automatically.

**Pull request merged**